Source/LogWriter.swift

```diff
@@ -25,18 +25,40 @@
 import Foundation
 import os
 
-/// The LogMessageWriter protocol defines a single API for writing a log message. The message can be written in any way
+/// The LogWriter protocol defines a single API for writing a log message. The message can be written in any way
 /// the conforming object sees fit. For example, it could write to the console, write to a file, remote log to a third
 /// party service, etc.
-public protocol LogMessageWriter {
-    func writeMessage(_ message: String, logLevel: LogLevel, modifiers: [LogMessageModifier]?)
+public protocol LogWriter {
+    func writeMessage(_ message: String, logLevel: LogLevel)
+    func writeMessage(_ message: Message, logLevel: LogLevel)
+}
+
+/// LogModifierWriter extends LogWriter to allow for standard writers that utilize MessageModifiers
+/// to transform the message before being output.
+public protocol LogModifierWriter: LogWriter {
+    var modifiers: [MessageModifier] { get }
+}
+
+public extension LogModifierWriter {
+    /// Apply all of the LogMessageModifiers to the incoming message and return a new message.
+    ///
+    /// - Parameters:
+    ///   - message:  Original message.
+    ///   - logLevel: Log level of message.
+    ///
+    /// - Returns: The result of executing all the modifiers on the original message.
+    public func modifyMessage(_ message: String, logLevel: LogLevel) -> String {
```

Comment:
I assumed inbound messages were restricted to a single log level, given that the public logging APIs set a single level (e.g. `log.debug()`), though I suppose one could create a custom logging level that set multiple bits, but I'm not sure that is a valid use case.